libraries/botbuilder-dialogs-adaptive/src/telemetryExtensions.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DialogManager } from 'botbuilder-dialogs';
+import { BotTelemetryClient } from 'botbuilder-core';
+
+/**
+ * The key to get or set telemetry client from turn state.
+ */
+export const telemetryClientKey = Symbol('telemetryClient');
+
+/**
+ * Extension methods for telemetry.
+ */
+export class TelemetryExtensions {
```

Comment:
The current naming protocol is to name the class with the Class/Type it is extending, in this case it would be "DialogManagerExtensions"

Reply:
@stevengum We have several helper classes (e.g. SkillExtensions, LanguageGenerationExtensions) to extend DialogManager, is it required to rename all of them to DialogManagerExtensions?

Reply:
In .NET, the extension methods and IntelliSense make it easy for developers to use these static helpers, so the class names haven't been as important. That said, @gabog and @johnataylor have indicated that the correct naming pattern is to use what is being extended as the class name.

From a porting perspective, using a singular `DialogManagerExtensions` helps in TypeScript, Python and Java as the developer only has to import `DialogManagerExtensions` and then use the appropriate static helper, versus having several different imports to access 1 or 2 helpers per import.

We may not fix this for the existing extensions in .NET (I'm not sure if this breaks binary or semver compat, though I don't believe it does), but going forward the consensus is to use `<ExtendedClass>Extensions` AFAIK.

Reply:
We can combine these helper classes into one in _botbuilder-dialogs-adaptive_ named `DialogManagerExtensions`. However, if we want to extend `DialogManager` from different packages, we may need to create multiple `DialogManagerExtensions` for each package, would that possibly an issue?

Reply:
It wouldn't because the ultimate reference name is defined by the consumer of the import:

```ts
import { DialogManagerExtensions as AdaptiveDMExtensions } from 'botbuilder-dialogs-adaptive';
import { DialogManagerExtensions } from 'botbuilder-dialogs-declarative';
import { DialogManagerExtensions as LUISDMExtensions }from 'botbuilder-ai';

```

Reply:
So is it okay to merge `LanguageGeneratorExtensions`, `ResourceExtensions`, `SkillExtensions` and `TelemetryExtensions` into one `DialogManagerExtensions` ? As they are all helper classes extending DialogManager.

Reply:
@Stevenic As we discussed before, we need to keep the name `TelemetryExtensions` as the c# sdk has been shipped already, right?

Reply:
FYI @steveinc @johnataylor @tomlm @gabog

I don't think the value gained here by having the same name for extension methods is comparable to having the same names for actual classes, constants and interfaces across languages.

In C#, a developer generally doesn't have to worry about importing the correct `static class` for extension methods because the methods are just "provided" via namespace and intellisense. The syntactic sugar makes it easy to consume extension methods (aka static helper methods) in C#.

If I'm `using Microsoft.Bot.Builder.Dialogs.Adaptive;`, then on my DialogManager instance I just call `.UseTelemetry()`.

___

In TS/JS I have to import the specific class (demonstrated in an [earlier comment](https://github.com/microsoft/botbuilder-js/pull/2513/files#r462874559)) and then reference the method I want:

```ts
const dm = new DialogManager(/* ... */);

import { LanguageGeneratorExtensions , SkillExtensions, TelemetryExtensions } from 'botbuilder-dialogs-adaptive';

TelemetryExtensions.useTelemetry(dm, telemetryClient);
SkillExtensions.useSkillClient(dm, skillClient);
LanguageGeneratorExtensions.useLanguageGeneration(dm, 'myMain.lg');
```

If we decide we want to have separate files, then the extension methods should just be turned into standalone static functions without a class.

```ts
const dm = new DialogManager(/* ... */);

import { useTelemetry , useSkillClient, useLanguageGeneration } from 'botbuilder-dialogs-adaptive';

useTelemetry(dm, telemetryClient);
useSkillClient(dm, skillClient);
useLanguageGeneration(dm, 'myMain.lg');
```

___

If the consensus is to keep stay with `TelemetryExtensions` (aka different classes, different files) instead of using `DialogManagerExtensions` (aka single class, single file), then I'm okay with doing it; but from a design POV it's just added superfluous work on our developers.

Reply:
I think it might be better to export standalone static functions instead of extension classes.

Reply:
I've asked @songcs to export `useTelemetry()` as a standalone static function and keep the filename unchanged.
If it is okay to implement extension methods in this way, I would send out another PR to rename the other extension methods as we don't need to wrap these methods in a class which is useless in javascript.

Reply:
@chon219 that sounds good to me, please send out another PR to rename/refactor the other extension methods.